libraries/Microsoft.Bot.Builder.Core.Extensions/CatchExceptionMiddleware.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class CatchExceptionMiddleware : IMiddleware
+    {
+        private readonly CatchExceptionHandler _handler;
+
+        public CatchExceptionMiddleware(CatchExceptionHandler handler)
```

Comment:
Is this a good way to do this?

I wonder if (for example) a virtual method would be cleaner.

Reply:
@cleemullins have you got an example of where we use a virtual method in such a way elsewhere?

Reply:
Updated the CatchExceptionMiddleware to take a generic type of Exception and swapped out the delegate for a func parameter.